libparsec/crates/platform_mountpoint/src/unix/mod.rs

```diff
@@ -0,0 +1,436 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use fuser::{
+    FileAttr, FileType, Filesystem, MountOption, ReplyAttr, ReplyCreate, ReplyData, ReplyDirectory,
+    ReplyEmpty, ReplyEntry, ReplyOpen, ReplyWrite, Request, Session, SessionUnmounter,
+};
+use libc::{ENAMETOOLONG, ENOENT};
+use libparsec_types::{anyhow, FileDescriptor};
+use std::{
+    ffi::OsStr,
+    path::{Path, PathBuf},
+    time::Duration,
+};
+
+use crate::{
+    error::MountpointError, EntryInfo, EntryInfoType, MountpointInterface, MountpointManager,
+    WriteMode,
+};
+
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct Counter(pub u64);
+
+impl Counter {
+    pub fn increment(&mut self) {
+        self.0 += 1;
+    }
+
+    pub fn decrement(&mut self, nlookup: u64) {
+        self.0 -= nlookup;
+    }
+
+    pub fn is_zero(&mut self) -> bool {
+        self.0 == 0
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct Inode(pub u64);
+
+impl From<usize> for Inode {
+    fn from(value: usize) -> Self {
+        Self(value as u64 + 1)
+    }
+}
+
+impl From<Inode> for usize {
+    fn from(value: Inode) -> Self {
+        value.0 as Self - 1
+    }
+}
+
+const BLOCK_SIZE: u64 = 512;
+const TTL: Duration = Duration::new(0, 0);
+const GENERATION: u64 = 0;
+
+impl From<EntryInfoType> for FileType {
+    fn from(value: EntryInfoType) -> Self {
+        match value {
+            EntryInfoType::Dir => Self::Directory,
+            EntryInfoType::File => Self::RegularFile,
+            EntryInfoType::ParsecEntryInfo => Self::RegularFile,
+        }
+    }
+}
+
+impl EntryInfo {
+    fn to_file_attr(&self, ino: u64) -> FileAttr {
+        let created = self.created;
+        let updated = self.updated;
+
+        let kind = match self.ty {
+            EntryInfoType::Dir => FileType::Directory,
+            EntryInfoType::File => FileType::RegularFile,
+            EntryInfoType::ParsecEntryInfo => FileType::RegularFile,
+        };
+
+        FileAttr {
+            ino,
+            size: self.size,
+            blocks: (self.size + BLOCK_SIZE - 1) / BLOCK_SIZE,
+            atime: updated.into(),
+            mtime: updated.into(),
+            ctime: updated.into(),
+            crtime: created.into(),
+            kind,
+            perm: 0,
+            nlink: 0,
+            uid: self.id.as_u128() as u32,
+            gid: self.id.as_u128() as u32,
+            rdev: 0,
+            blksize: BLOCK_SIZE as u32,
+            flags: 0,
+        }
+    }
+}
+
+impl Filesystem for MountpointManager {
+    fn lookup(&mut self, _req: &Request, parent: u64, name: &OsStr, reply: ReplyEntry) {
+        let parent = self.get_path(Inode(parent));
```

Comment:
rename variable so it does not clashes with parameter

Reply:
can get_path fail ? shouldn't we check if exists?

Reply:
> can get_path fail ? shouldn't we check if exists ?

This is ok for the memfs

Reply:
> rename variable so it does not clashes with parameter
Done